flags.go

```diff
@@ -341,6 +345,69 @@ type flagStorage struct {
 	DebugMutex      bool
 }
 
+const GCSFUSE_PARENT_PROCESS_DIR = "gcsfuse-parent-process-dir"
+
+// 1. Returns the same filepath in case of absolute path or empty filename.
+// 2. For relative path with . or .. or nothing, it resolves smartly the path
+// in case of child proces it resolves with respect to GCSFUSE_PARENT_PROCESS_DIR
+// otherwise with respect to the current working dir.
+// 3. For relative path starting with ~, it resolves with respect to home dir.
+func getResolvedPath(filePath string) (resolvedPath string, err error) {
+	if filePath == "" || path.IsAbs(filePath) {
+		resolvedPath = filePath
+		return
+	}
+
+	// Relative path starting with tilda (~)
+	if strings.HasPrefix(filePath, "~/") {
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			return "", fmt.Errorf("Fetch home dir: [%w]", err)
+		}
+		return filepath.Join(homeDir, filePath[2:]), err
+	}
+
+	// Means - relative path starting with ., .. or nothing.
+	gcsfuseParentProcessDir, _ := os.LookupEnv(GCSFUSE_PARENT_PROCESS_DIR)
+	gcsfuseParentProcessDir = strings.TrimSpace(gcsfuseParentProcessDir)
+	if gcsfuseParentProcessDir == "" {
+		return filepath.Abs(filePath)
+	} else {
+		return filepath.Join(gcsfuseParentProcessDir, filePath), err
+	}
+}
+
+// This method resolves path in the context dictionary.
+func resolvePathForTheFlagInContext(flagKey string, c *cli.Context) (err error) {
+	flagValue := c.String(flagKey)
+	resolvedPath, err := getResolvedPath(flagValue)
+	if flagValue == resolvedPath || err != nil {
+		return
+	}
+	c.Set(flagKey, resolvedPath)
+	logger.Infof("Value of [%s] resolved from [%s] to [%s]\n",
+		flagKey, flagValue, resolvedPath)
+
+	return
+}
+
+// For parent process: it only resolves the path with respect to home folder.
+// For child process (when --foreground flag is disabled): it resolves
+// the path relative to both current directory and home directory
+func resolvePathForTheRequiredFlagInContext(c *cli.Context) (err error) {
+	err = resolvePathForTheFlagInContext("log-file", c)
+	if err != nil {
+		return fmt.Errorf("resolving for log-file: [%w]", err)
```

Comment:
starting with capital letter and remove square brackets.
same for line 405

Reply:
Getting warning "Error string should not be capitalized or end with punctuation mark ".